src/hitbox.h

```diff
@@ -1,13 +1,29 @@
-#pragma once 
+#pragma once
 #include <SDL2/SDL.h>
 
-class Hitbox { 
-public: 
-	Hitbox(double x_pos, double y_pos, double width, double height); 	
-	Hitbox(SDL_Rect hitbox); 
-	bool is_overlapping(const Hitbox& other_hitbox);
-	const SDL_Rect& hitbox() const;
+struct Circle
+{
+    int x, y;
+    int r;
+};
+
+class Hitbox {
+public:
+    Hitbox(double x_pos, double y_pos, double width, double height);
+    Hitbox(double x_pos, double y_pos, double radius);
+    Hitbox(SDL_Rect hitbox);
+    Hitbox(Circle hitbox);
+
+    bool is_overlapping(Hitbox& other_hitbox);
+    const SDL_Rect& hitbox() const;
+    const Circle& hitbox_circle() const;
```

Comment:
I'd recommend returning pointers here instead, that way you can return a NULL value instead.